authors/apps/profiles/views.py

```diff
@@ -0,0 +1,45 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import generics
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.permissions import (
+    IsAuthenticated,
+)
+
+from .models import Profile
+from .serializers import (
+    UserProfileSerializer,
+    ProfileUpdateSerializer
+)
+from authors.apps.utilities.messages import error_messages
+from .renderers import UserProfileJSONRenderer
+
+from authors.apps.utilities.custom_permissions.permissions import if_owner_permission
+
+
+class UserProfileView(generics.RetrieveAPIView):
+    """ Fetches and displays the details
+    of a user profile to the currently
+    logged in person
+    """
+    permission_classes = (IsAuthenticated,)
+    serializer_class = UserProfileSerializer
+    renderer_classes = (UserProfileJSONRenderer, )
+
+    def get_object(self, *args, **kwargs):
+        username = self.kwargs.get("username")
+        return get_object_or_404(Profile, user__username=username)
+
+
+class UserProfileUpdateView(generics.UpdateAPIView):
+    """ Allows the currently logged in user
+    to edit their user profile
+    possible edittable fields include,
+    first_name, last_name, bio
+    and image """
+    serializer_class = ProfileUpdateSerializer
```

Comment:
@Rhytah , the update  response is not proper properly rendered
This is returned
```
{
    "first_name": null,
    "last_name": null,
    "bio": null,
    "image": "url",
    "date_of_birth": null
}
```

This is expected
```
{
    "profile": {
        "username": "Michealkv",
        "first_name":"fnanem",
        "last_name": "Peter"
    }
}
```

And a renderer for it